### DISCUSSION TITLE
Registering Objects

### DISCUSSION THREAD

**Discussion by Leni-Yeo** [idaholab/moose#16833] [Q&A Getting Started] [answered]
Hello, I am a student and new user for MOOSE. I was doing the workshop and trying the registration for the "DarcyPressure" kernel. The step do not specifically tell where and how to specifically register. It is initially registered under "DarcyThermoMechApp" in the workshop section, but I keep getting a " 'DarcyPressure' is not a registered object " error. I tried registering under "MooseApp" and got the same trouble. I was just trying to run the tutorial input file step2.i . I was told that I do not need to create a  new application to run it (Which I am not sure either). It would be helpful if you could provide locations where to save the .h and .C files or the general steps to do in the case I need to save/register a new object. Thank you.

**Comment by aeslaughter** [comment, marked as answer]
We have detailed instructions that might help.
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html

**Reply by aeslaughter** [reply]
I recommend going through each step carefully. You will find some help specific to your question in Step 5 of this new tutorial.

**Reply by Leni-Yeo** [reply]
Thank you, that worked!
